go/ql/src/Security/CWE-078/examples/CommandInjectionGood3.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"os/exec"
+	"strings"
+)
+
+func handler(req *http.Request) {
+	repoURL := req.URL.Query()["repoURL"][0]
+	outputPath := "/tmp/repo"
+
+	// Sanitize the repoURL to ensure it does not start with "--"
+	if strings.HasPrefix(repoURL, "--") {
+		log.Fatal("Invalid repository URL")
+		return
+	}
```

Comment:
Surely you need a command execution guarded by the sanitizer?
```suggestion
	} else {
	  cmd := exec.Command("git", "clone", repoURL, outputPath)
	  err := cmd.Run()
	    	if err != nil {
     		log.Fatal(err)
     	}
  }
```